frame/tx-pause/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::pause())]
+		pub fn pause(origin: OriginFor<T>, full_name: FullNameOf<T>) -> DispatchResult {
+			T::PauseOrigin::ensure_origin(origin)?;
+
+			Self::do_pause(full_name).map_err(Into::into)
+		}
+
+		/// Un-pause a call.
+		///
+		/// Can only be called by [`Config::UnpauseOrigin`].
+		/// Emits an [`Event::CallUnpaused`] event on success.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::unpause())]
+		pub fn unpause(origin: OriginFor<T>, ident: FullNameOf<T>) -> DispatchResult {
+			T::UnpauseOrigin::ensure_origin(origin)?;
+
+			Self::do_unpause(ident).map_err(Into::into)
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub(crate) fn do_pause(ident: FullNameOf<T>) -> Result<(), Error<T>> {
+		Self::ensure_can_pause(&ident)?;
+		PausedCalls::<T>::insert(&ident, ());
+		Self::deposit_event(Event::CallPaused { full_name: ident });
+
+		Ok(())
+	}
+
+	pub(crate) fn do_unpause(ident: FullNameOf<T>) -> Result<(), Error<T>> {
+		Self::ensure_can_unpause(&ident)?;
+		PausedCalls::<T>::remove(&ident);
+		Self::deposit_event(Event::CallUnpaused { full_name: ident });
+
+		Ok(())
+	}
+
+	/// Return whether this call is paused.
+	pub fn is_paused_unbound(pallet: Vec<u8>, call: Vec<u8>) -> bool {
+		let pallet = PalletNameOf::<T>::try_from(pallet);
+		let call = CallNameOf::<T>::try_from(call);
+
+		match (pallet, call) {
+			(Ok(pallet), Ok(call)) => Self::is_paused(&(pallet, call)),
+			_ => T::PauseTooLongNames::get(),
+		}
+	}
+
+	/// Return whether this call is paused.
+	pub fn is_paused(full_name: &FullNameOf<T>) -> bool {
+		if T::WhitelistedCalls::contains(full_name) {
+			return false
+		}
+
+		<PausedCalls<T>>::contains_key(full_name)
+	}
+
+	/// Ensure that this call can be paused.
+	pub fn ensure_can_pause(full_name: &FullNameOf<T>) -> Result<(), Error<T>> {
+		// SAFETY: The `TxPause` pallet can never pause itself.
+		if full_name.0.as_ref() == <Self as PalletInfoAccess>::name().as_bytes().to_vec() {
+			return Err(Error::<T>::Unpausable)
+		}
+
+		if T::WhitelistedCalls::contains(&full_name) {
+			return Err(Error::<T>::Unpausable)
+		}
+		if Self::is_paused(&full_name) {
+			return Err(Error::<T>::IsPaused)
+		}
+		Ok(())
+	}
+
+	/// Ensure that this call can be un-paused.
+	pub fn ensure_can_unpause(full_name: &FullNameOf<T>) -> Result<(), Error<T>> {
+		if Self::is_paused(&full_name) {
+			// SAFETY: Everything that is paused, can be un-paused.
+			Ok(())
+		} else {
+			Err(Error::IsUnpaused)
+		}
+	}
+}
+
+impl<T: pallet::Config> Contains<<T as frame_system::Config>::RuntimeCall> for Pallet<T>
+where
+	<T as frame_system::Config>::RuntimeCall: GetCallMetadata,
+{
+	/// Return whether the call is allowed to be dispatched.
+	fn contains(call: &<T as frame_system::Config>::RuntimeCall) -> bool {
+		let CallMetadata { pallet_name, function_name } = call.get_call_metadata();
+		!Pallet::<T>::is_paused_unbound(pallet_name.into(), function_name.into())
```

Comment:
Why not using the bounded version `is_paused` instead?

Reply:
Not possible, since the `pallet_name` and `function_name` are unbounded `&'static str`.